src/ListView/features/CreatePage/CreatePageModal.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState } from "react";
+
+import { Modal, Notification } from "@sailthru/stui-components";
+import * as pagesApi from "../../../core/pagesApi";
+import { CreatePageForm } from "./CreatePageForm";
+
+function CreatePageModal({ title, mode, setDisplayModal }) {
+  const [state, setState] = useState({});
+  const [error, setError] = useState(false);
+  const [isDuplicateName, setIsDuplicateName] = useState(false);
+  useEffect(() => {
+    setIsDuplicateName(false);
+  }, [isDuplicateName]);
+  const [isNameChanged, setIsNameChanged] = useState(false);
+
+  function handleCreate() {
+    pagesApi
+      .createPage({ ...state, mode })
+      .then(function (response) {
+        if (response._data.editor_url) {
+          window.location.href = `${window.location.origin}${response._data.editor_url}`;
+        } else {
+          // this should never happen
+          setError("Something went wrong and we could not load the page.");
+          console.log("An error occurred redirecting to the new page");
+          window.location.href = `${window.location.origin}/hosted-pages-list`;
+        }
+        return response._data;
+      })
+      .catch(function (serverError) {
+        let duplicateNameError = "There is already a page named";
+        let serverErrorMessage = serverError.error_message;
+        if (
+          serverErrorMessage &&
+          serverErrorMessage.includes(duplicateNameError)
```

Comment:
nit-pick: I don't think the variable name helps with clarity here
```suggestion
        let serverErrorMessage = serverError.error_message;
        if (
          serverErrorMessage &&
          serverErrorMessage.includes("There is already a page named")
```

Reply:
removed this variable